client/app/pages/index.js

```diff
@@ -0,0 +1,53 @@
+import { flatten } from "lodash";
+
+import adminJobsRoutes from "./admin/Jobs";
+import adminOutdatedQueriesRoutes from "./admin/OutdatedQueries";
+import adminSystemStatusRoutes from "./admin/SystemStatus";
+import alertRoutes from "./alert/Alert";
+import alertsListRoutes from "./alerts/AlertsList";
+import dashboardListRoutes from "./dashboards/DashboardList";
+import dashboardRoutes from "./dashboards/DashboardPage";
+import publicDashboardRoutes from "./dashboards/PublicDashboardPage";
+import dataSourcesListRoutes from "./data-sources/DataSourcesList";
+import editDataSourceRoutes from "./data-sources/EditDataSource";
+import destinationsListRoutes from "./destinations/DestinationsList";
+import editDestinationRoutes from "./destinations/EditDestination";
+import groupsListRoutes from "./groups/GroupsList";
+import groupsDataSourcesRoutes from "./groups/GroupDataSources";
+import groupsMembersRoutes from "./groups/GroupMembers";
+import homeRoutes from "./home/Home";
+import querySourceRoutes from "./queries/QuerySource";
+import queryViewRoutes from "./queries/QueryView";
+import visualizationEmbedRoutes from "./queries/VisualizationEmbed";
+import queriesListRoutes from "./queries-list/QueriesList";
+import querySnippetsRoutes from "./query-snippets/QuerySnippetsList";
+import organizationSettingsRoutes from "./settings/OrganizationSettings";
+import userProfileRoutes from "./users/UserProfile";
+import usersListRoutes from "./users/UsersList";
+
+export default flatten([
+  adminJobsRoutes,
+  adminOutdatedQueriesRoutes,
+  adminSystemStatusRoutes,
+  alertRoutes,
+  alertsListRoutes,
+  dashboardListRoutes,
+  dashboardRoutes,
+  publicDashboardRoutes,
+  dataSourcesListRoutes,
+  editDataSourceRoutes,
+  destinationsListRoutes,
+  editDestinationRoutes,
+  groupsListRoutes,
+  groupsDataSourcesRoutes,
+  groupsMembersRoutes,
+  homeRoutes,
+  queriesListRoutes,
+  queryViewRoutes,
+  querySourceRoutes,
+  visualizationEmbedRoutes,
+  querySnippetsRoutes,
+  organizationSettingsRoutes,
+  usersListRoutes,
+  userProfileRoutes,
```

Comment:
Forgot to reply to this comment. Why I ended up using explicit imports: if you remember that issue with parameters classes (circular references) - it happened because we removed `require.resolve` for services and started importing them explicitly, and Webpack did a deeper optimization of that files which revealed that bug (the fact it worked before was just an accident). I googled about that a bit and found, that for dynamic and implicit imports Webpack creates so-called "context module", and for that module type some optimizations are unavailable because Webpack cannot guess if that module has any side effect. When using explicit imports, Weback can properly analyze code and perform better optimization of the bundle.

Reply:
:+1: good find.

Not something to spend on time now, but it's interesting to see how Next.js handles this. Because they have dynamic imports and produce optimized builds.

Reply:
It does not mean that with dynamic imports bundle wasn't optimized. But, for example, with explicit imports Webpack can merge modules, mangle exports (because now it exactly knows where they're used), etc.